components/kubernetes/kind.go

```diff
@@ -108,3 +110,70 @@ func NewKindCluster(env config.CommonEnvironment, vm *remote.Host, resourceName,
 		return nil
 	}, opts...)
 }
+
+func NewLocalKindCluster(env config.CommonEnvironment, resourceName, kindClusterName string, kubeVersion string, opts ...pulumi.ResourceOption) (*Cluster, error) {
+	return components.NewComponent(env, resourceName, func(clusterComp *Cluster) error {
+		opts = utils.MergeOptions[pulumi.ResourceOption](opts, pulumi.Parent(clusterComp))
+		commonEnvironment := env
+
+		user, err := user.Current()
+		if err != nil {
+			return err
+		}
+
+		kindVersionConfig, err := getKindVersionConfig(kubeVersion)
+		if err != nil {
+			return err
+		}
+
+		runner := command.NewLocalRunner(env, command.LocalRunnerArgs{
+			User:      user.Username,
+			OSCommand: command.NewUnixOSCommand(),
+		})
+
+		clusterConfigFilePath := fmt.Sprintf("/tmp/kind-cluster-%s.yaml", kindClusterName)
+		clusterConfig, err := runner.Command("kind-config", &command.Args{
+			Create: pulumi.Sprintf("cat - | tee %s > /dev/null", clusterConfigFilePath),
+			Delete: pulumi.Sprintf("rm -f %s", clusterConfigFilePath),
+			Stdin:  pulumi.String(kindClusterConfig),
+		}, opts...)
+		if err != nil {
+			return err
+		}
+
+		nodeImage := fmt.Sprintf("%s:%s", kindNodeImageRegistry, kindVersionConfig.nodeImageVersion)
```

Comment:
Actually, I think this is not that easy to create an interface shared by the two runners because the `Command` method uses two different stuct `remote.Command` and `local.Command` which does not share an interface. I think we could achieve this by wrapping `local.Command` and `remote.Command` in a struct that implements a `Command` interface but that looks like a big refactor